postgresql/src/main/java/org/apache/zeppelin/postgresql/PostgreSqlInterpreter.java

```diff
@@ -123,10 +145,16 @@ public void open() {
 
       jdbcConnection = DriverManager.getConnection(url, user, password);
 
+      Set<String> keywordsCompletions = SqlCompleter.getSqlKeywordsCompletions(jdbcConnection);
+      Set<String> dataModelCompletions =
+          SqlCompleter.getDataModelMetadataCompletions(jdbcConnection);
+      SetView<String> allCompletions = Sets.union(keywordsCompletions, dataModelCompletions);
+      sqlCompleter = new SqlCompleter(allCompletions, dataModelCompletions);
+
       exceptionOnConnect = null;
       logger.info("Successfully created psql connection");
 
-    } catch (ClassNotFoundException | SQLException e) {
+    } catch (ClassNotFoundException | SQLException | IOException e) {
```

Comment:
In general the Interpreter has little or no control on when the open()/close() methods are called. The Zeppelin frameworks calls open once through the class loader  and the lazy interpreters. 
So the strategy is to create and keep the JDBC connection for the duration of the interpreter life and create prepared-statements per each SQL execute query.  A future improvement would be to employ a connection pool and then open/close the jdbcConnection on each query execution. 
Particularly in this method (PostgreSqlInterpreter#opne()) we do not  want to close the connection on any other errors but connection related. For example if something gets wrong with the `SqlCompleter` initialization this should not close the connection. 
One possible improvement would be to have a nested `try/catch` block  around the 
```java
Class.forName(driverName);
jdbcConnection = DriverManager.getConnection(url, user, password);
```
and try to clean/close the connection if failure happens there. But if the failure happens there most likely the connection has not been created on the first place?
@hsaputra what do you think? Am i missing something important that can lead to resource leaking?